app/views/handlers/feedback.py

```diff
@@ -254,8 +259,8 @@ def __init__(
         metadata: Mapping[str, Union[str, int, list]],
         response_metadata: Mapping[str, Union[str, int, list]],
         schema: QuestionnaireSchema,
-        case_id: str,
-        submission_language_code: Optional[str],
+        case_id: METADATA_MAPPING_TYPE,
+        submission_language_code: METADATA_MAPPING_TYPE,
```

Comment:
Similar to previous comment, I don't think we are able to cast more specific type on these, because they are given more generic type in `authenticator.py`:
```
def _create_session_data_from_metadata(
    metadata: Mapping[str, Union[str, int, list]]
```
so they could be any of str, int, list types as they are extracted using `metadata.get()`.

Reply:
We need to make the initial metadata `Mapping[str, Any]` so that we can be explicit. We could also use typed dicts.

Reply:
Done now!